modules/fargene/main.nf

```diff
@@ -0,0 +1,64 @@
+// Import generic module functions
+include { initOptions; saveFiles; getSoftwareName; getProcessName } from './functions'
+
+params.options = [:]
+options        = initOptions(params.options)
+
+def VERSION = '0.1'
+
+process FARGENE {
+    tag "$meta.id"
+    label 'process_low'
+    publishDir "${params.outdir}",
+        mode: params.publish_dir_mode,
+        saveAs: { filename -> saveFiles(filename:filename, options:params.options, publish_dir:getSoftwareName(task.process), meta:meta, publish_by_meta:['id']) }
+
+    conda (params.enable_conda ? "bioconda::fargene=0.1" : null)
+    if (workflow.containerEngine == 'singularity' && !params.singularity_pull_docker_container) {
+        container "https://depot.galaxyproject.org/singularity/fargene:0.1--py27h21c881e_4"
+    } else {
+        container "quay.io/biocontainers/fargene:0.1--py27h21c881e_4"
+    }
+
+    input:
+    tuple val(meta), path(input) //input may be fasta (for genomes or longer contigs) or paired-end fastq (for metagenome), the latter in addition with --meta flag
+    val(hmm_model)
+
+    output:
+    path("*.log"),                                                                                    emit: log
+    path("${prefix}/results_summary.txt"),                                                                      emit: txt
+    tuple val(meta), path("${prefix}/hmmsearchresults/*.out"),                                  optional: true, emit: hmm
+    tuple val(meta), path("${prefix}/predictedGenes/predicted-orfs.fasta"),                     optional: true, emit: orfs
+    tuple val(meta), path("${prefix}/predictedGenes/predicted-orfs-amino.fasta"),               optional: true, emit: orfs_amino
+    tuple val(meta), path("${prefix}/predictedGenes/retrieved-contigs.fasta"),                  optional: true, emit: contigs
+    tuple val(meta), path("${prefix}/predictedGenes/retrieved-contigs-peptides.fasta"),         optional: true, emit: contigs_pept
+    tuple val(meta), path("${prefix}/predictedGenes/*filtered.fasta"),                          optional: true, emit: filtered
+    tuple val(meta), path("${prefix}/predictedGenes/*filtered-peptides.fasta"),                 optional: true, emit: filtered_pept
+    tuple val(meta), path("${prefix}/retrievedFragments/all_retrieved_*.fastq"),                optional: true, emit: fragments
+    tuple val(meta), path("${prefix}/retrievedFragments/retrievedFragments/trimmedReads/*.fasta"), optional: true, emit: trimmed
```

Comment:
the indent should be fixed here, I'd commit the suggestion myself but the browser editor doesn't display the suggestion clearly